internals/overlord/overlord.go

```diff
@@ -258,6 +265,15 @@ func (o *Overlord) StartUp() error {
 		return nil
 	}
 	o.startedUp = true
+
+	var err error
```

Comment:
In snapd [this is inside](https://github.com/snapcore/snapd/blob/5d02ea78acef7bfb8bd2ce50c7874c895acde832/overlord/overlord.go#L362) an `if deviceMgr != nil` block. We don't have a device manager in Pebble. Should this still apply? What is "start of operation time"?

Reply:
I believe that snapd uses this time to accommodate seeding. The device manager uses 'seed-time' as the start of operation time if available, i.e. the time when seeding has finished and the state has been [marked as 'seeded'](https://github.com/snapcore/snapd/blob/5d02ea78acef7bfb8bd2ce50c7874c895acde832/overlord/devicestate/handlers.go#L185). If 'seed-time' is not available the device manager falls back to using time.Now() at the time of the call.

There is no relevant use case in Pebble AFAIK (yet). The start of operation time is always the time when the overlord's StartUp was called for the first time. I left this logic only to keep the state package unchanged as [state.Prune](https://github.com/snapcore/snapd/blob/5d02ea78acef7bfb8bd2ce50c7874c895acde832/overlord/state/state.go#L440) accepts the start of operation time as a parameter to decide whether a change needs to be pruned based on the spawn/start-of-operation time (i.e. whether there are existing changes that might have been spawned before the start of operation time; those do not exist in Pebble's case as all its changes would be spawned after the managers have their startup logic completed).

Reply:
@pedronis thank you. Yes, this is what I've figured from reading snapd. I wanted this merge to have least possible deviations from the interface and logic of the state package. It is correct that Pebble does not have such changes, so the start of operation will always be the first start time.

@benhoyt we could do as Samuele suggested here and just pass a zero-time to Prune and comment in details why we do that. This will rid us of storing the start of operation time similarly to snapd as I described above. However, given that state.Prune expects the start of operation semantically it could make sense to pass what it actually expects for better readability and, perhaps, making future merges easier. WDYT?

Reply:
Yeah, the latter sounds right to make it semantically correct and future merges easier -- is that what you have in the PR now?

Reply:
@benhoyt yes.